### PR TITLE
Implement linear-gradient, radial-gradient and dummy-pattern in SVGGraphics.

### PR DESCRIPTION
**Pull request by terurou**

### CODE REVIEW

File: src/display/svg.js [hunk 1]
```diff
@@ -936,6 +946,91 @@ SVGGraphics = (function SVGGraphicsClosure() {
       this.current.tspan = this.svgFactory.createElement('svg:tspan');
       this.current.xcoords = [];
     },
+    setStrokeColorN: function SVGGraphics_setStrokeColorN(args) {
+      this.endPathcurrent.strokeColor = this._makeColorN_Pattern(args);
```
Comment (timvandermeij) [inline src/display/svg.js]: Shouldn't this be `this.current.strokeColor`?

Comment (terurou) [reply src/display/svg.js]: Thank you! I fixed this mistake.

File: src/display/svg.js [hunk 2]
```diff
@@ -936,6 +946,91 @@ SVGGraphics = (function SVGGraphicsClosure() {
...
+        return null;
+      }
+      return this._makeShadingPattern(args);
+    },
+    _makeShadingPattern: function SVGGraphics_makeShadingPattern(args) {
+      switch (args[0]) {
+        case 'RadialAxial':
+          var shadingId = 'shading' + shadingCount++;
+          var colorStops = args[2];
+          var gradient = this.svgFactory.createElement('svg:linearGradient');
```
Comment (timvandermeij) [inline src/display/svg.js]: In the case of a radial gradient, I think we should use `svg:radialGradient` instead. According to https://developer.mozilla.org/en-US/docs/Web/SVG/Element/linearGradient, there are no `cx` and `cy` attributes, whereas `radialGradient` has those.

Comment (terurou) [reply src/display/svg.js]: Thanks very much. I fixed it.
Maybe I mistaked when I refactored for the eslint warnings...

**Review by timvandermeij — APPROVED** [review]

**Comment by timvandermeij** [comment]
Thank you for implementing this!

**Comment by Snuffleupagus** [comment]
@terurou Please remember to actually squash fix-up commits in the future, as was asked for above, since the result here was that this PR landed with broken intermediate state meaning that e.g. future bisecting will now be *a lot* more difficult.

**Comment by timvandermeij** [comment]
You're right. I also forgot to choose the "squash and commit" option on GitHub.

**Comment by terurou** [comment]
@Snuffleupagus Thank you for your advice. I will be careful from the next time.

**Pull request merged**